src/components/WalletAssets.vue

```diff
@@ -207,8 +204,44 @@ export default class WalletAssets extends Mixins(LoadingMixin, FormattedAmountMi
     }
   }
 
-  mounted(): void {
-    this.assetList = api.assets.accountAssets;
+  filterAssets(filters: WalletAssetFilters): void {
+    if (filters.option === WalletFilteringOptions.ALL) {
+      this.assetList = api.assets.accountAssets.filter((asset) => this.applyFilterOptions(asset, filters));
+    }
+
+    if (filters.option === WalletFilteringOptions.TOKEN) {
+      const tokens = api.assets.accountAssets.filter((asset) => !api.assets.isNft(asset));
+      this.assetList = tokens.filter((asset) => this.applyFilterOptions(asset, filters));
+    }
+
+    if (filters.option === WalletFilteringOptions.NFT) {
+      const nfts = api.assets.accountAssets.filter((asset) => api.assets.isNft(asset));
+      this.assetList = nfts.filter((asset) => this.applyFilterOptions(asset, filters));
+    }
+  }
+
+  applyFilterOptions(asset: AccountAsset, filters: WalletAssetFilters) {
+    // filter
+    const showWhitelistedOnly = filters.verifiedOnly;
+    const hideZeroBalance = filters.zeroBalance;
+
+    // asset
+    const isWhitelisted = api.assets.isWhitelist(asset, this.whitelist);
+    const notHaveZeroBalance = asset.balance.total !== '0';
+
+    if (!isWhitelisted && showWhitelistedOnly) {
+      return false;
+    }
+
+    if (hideZeroBalance && !notHaveZeroBalance) {
+      return false;
+    }
+
+    return true;
+  }
+
+  updated(): void {
+    !this.assetList.length ? (this.showEmptyListText = true) : (this.showEmptyListText = false);
```

Comment:
`this.showEmptyListText = !this.assetList.length`
btw, do we really need it in `updated` hook?

Reply:
why not? need to check every time filter was clicked 
cleaner than figuring out somewhere else